### PR TITLE
Fix bugs in GitHub Actions

### PR DESCRIPTION
**Pull request by LorenzE**
- Fix documentation upload
- Add Github Actions badges to README.md
- Add new job to change the tag of the development build release
- Fix problem where the dev_build release tag was not pointing to the last commit
- Merge push scripts to devbuilds.yml file
- Rename some jobs
- Bump up Qt version in MinimalVersion job to 5.6

### CODE REVIEW

**Comment by buildqa** [comment]
Wow - lot's of good work on devbuilds.yml to integrate build, tag and release steps.

**Comment by LorenzE** [comment]
@buildqa Thx for the swift review. I think you closed the PR though. I reopened it. Can you approve once again?

**Comment by buildqa** [comment]
OK, sorry I closed and commented instead of just commenting.

**Comment by LorenzE** [comment]
@buildqa Can you give us another review?

File: .github/workflows/devbuilds.yml [hunk 1]
```diff
@@ -1,14 +1,47 @@
...
+    - name: Update dev_build tag and release
+      env:
+        GITHUB_USER: LorenzE
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        git config --global user.email lorenzesch@hotmail.com
+        git config --global user.name $GITHUB_USER
+        # Delete current dev_build release remotley. 
+        # Must be done before deleting the remote tag associated with the dev_build release. 
+        # This prevents a draft release to be left over after we delete the tag remotley.
```
Comment (buildqa) [inline .github/workflows/devbuilds.yml]: this is all much easier to read/follow

File: .github/workflows/devbuilds.yml [hunk 2]
```diff
@@ -36,28 +69,81 @@ jobs:
...
+      run: |
+        7z a mne-cpp-windows-x86_64.zip ./bin
+    - name: Deploy binaries with dev_build release on Github
+      uses: svenstaro/upload-release-action@v1-release
+      with:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+        file: mne-cpp-windows-x86_64.zip
+        asset_name: mne-cpp-windows-x86_64.zip
+        tag: dev_build
+        overwrite: true
```
Comment (buildqa) [inline .github/workflows/devbuilds.yml]: Nicely readable - not clear to me if deploy still uses the Qt deploy commands

Comment (LorenzE) [reply .github/workflows/devbuilds.yml]: What do you mean by Qt deploy commands?

File: .github/workflows/linux_pr.yml [hunk 3]
```diff
@@ -18,12 +18,11 @@ jobs:
...
         modules: qtcharts
     - name: Configure and compile MNE-CPP
-      run: |
-        qmake -query    
+      run: |    
         qmake -r MNECPP_CONFIG+=noTests
         make -j2 
 
-  ReleaseQtMinimum:
+  QtMinimum:
```
Comment (buildqa) [inline .github/workflows/linux_pr.yml]: There are entries for QtLatest (above) and QtMinimum here both with run-on: ubuntu-latest   Above there is Linux: then indented, runs-on: Ubuntuu 16-04.   I wish there was a way to define as variables QtLatest and QtMinimum and then subsequently use them as was follows "runs-on".   Or I would assume QtMinimum shouls always be Ubuntu 16-04

Comment (LorenzE) [reply .github/workflows/linux_pr.yml]: Are you maybe referring to this feature, see [here](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix).

Comment (buildqa) [reply .github/workflows/linux_pr.yml]: OK - looks ike the idea is to allow setup of an OS matrix and then different OS can run on a specific node.

**Pull request merged**